lib/View/courseoverview_screen.dart

```diff
@@ -0,0 +1,144 @@
+import 'package:flutter/material.dart';
+import 'package:gocast_mobile/View/utils/custom_bottom_nav_bar.dart';
+import 'package:gocast_mobile/View/mycourses_screen.dart';
+import 'package:gocast_mobile/View/publiccourses_screen.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:gocast_mobile/View/utils/viewall_button_view.dart';
+import 'settings_screen.dart';
+import 'package:gocast_mobile/View/utils/course_card_view.dart';
+
+final currentIndexProvider = StateProvider<int>((ref) => 0);
+
+class CourseOverview extends ConsumerWidget {
+  const CourseOverview({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
```

Comment:
Extract Widgets: Large build methods are wuite difficult to read and maintain. Consider breaking it down intoo smaller, named widget/methods.

Reply:
done